### PR TITLE
Updated deprecated gh-script action to node 20 variant

### PR DESCRIPTION
**Pull request by lanej0**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Noticed a [deprecation warning](https://github.com/woocommerce/woocommerce/actions/runs/8706622863) on one of the actions used in `ci.yml`.  This small PR updates the version of that action.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure CI passes
2. Confirm that deprecation warning is no longer shown on Summary screen

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/tests/e2e-pw/tests/basic.spec.js [hunk 1]
```diff
@@ -21,6 +21,7 @@ test( 'Load the home page', async ( { page } ) => {
 test( 'Load wp-admin as admin', async ( { page } ) => {
 	await page.context().clearCookies();
 	await page.goto( '/wp-admin' );
+	await page.waitForLoadState( 'domcontentloaded' );
```
Comment (adimoldovan) [inline plugins/woocommerce/tests/e2e-pw/tests/basic.spec.js]: Is this change intentional for the PR or maybe you didn't want to push it?

Comment (lanej0) [reply plugins/woocommerce/tests/e2e-pw/tests/basic.spec.js]: Just pushed it to trigger ci.yml to make sure everything was still okay.  I'll revert it before merging.

**Review by adimoldovan — APPROVED** [review]
LGTM! 👍🏼
